### ISSUE TITLE
incorrect types reported in error for non-matching types in non-equi join

### ISSUE THREAD

**Issue by tdhock** [Rdatatable/data.table#6069]
Hi! I noticed that the error message for non-equi join is incorrect, when types do not match
```r
point.dt <- data.table(data.i=1L)
seg.dt <- data.table(first.i=1L, last.i=2)
seg.dt[point.dt, on=.(first.i<=data.i,last.i>=data.i)]
# Error in bmerge(i, x, leftcols, rightcols, roll, rollends, nomatch, mult,  : 
#   typeof x.first.i (integer) != typeof i.data.i (double)
```
The type of i.data.i is integer, not double. The error message should be changed.

**Comment by venom1204** [comment]
can you guide me a little bit regarding this issue 
do we only need to change the bmerge.r file in the R folder @tdhock .

**Comment by ben-schwen** [comment]
@tdhock it seems that this issue has been fixed by #6603

**Comment by tdhock** [comment]
right, now the merge works (no error)
```r
> seg.dt[point.dt, on=.(first.i<=data.i,last.i>=data.i)]
   first.i last.i
     <int>  <int>
1:       1      1
```
